### ISSUE TITLE
Package installer crashes when installing Python.org Python 3.10.5

### ISSUE THREAD

**Issue by hamishmb** [darlinghq/darling#1202]
**Expected Result**
Python 3.10.5 installs successfully (worked previously with Darling using the LKM).

**Actual Result**
Installation fails with lots of "Resource Temporarily Unavailable" errors, followed by a Python stack trace.

**Steps To Reproduce**
Consistently crashes the same way when trying to install Python 3.10.5 the usual way with the package installer.

**System Information**

| Software | Version |
| Debian | 11  |
| Linux Kernel | 5.10.0 |
| Darling | e385bb9d9898987119c6514d01d403204e00704d |

I don't know how to debug this but I'd be happy to try if someone's willing to hand-hold me through the process.

**Comment by facekapow** [comment]
I'm unable to reproduce this locally on the latest commit; both 3.10.5 and the newest release, 3.10.6, install and run without issue. Could you try this again in a new prefix? (set `DPREFIX` to some path before running Darling).

If the issue persists, one thing you can try is running it with `xtrace`: just prepend `xtrace ` to the beginning of the command (e.g. `xtrace echo hello`).

Just to be sure, you're installing the package using `installer -pkg python-3.10.5-macos11.pkg -target /`, right?

**Comment by hamishmb** [comment]
Yes, I'll rebuild and try again - there have been quite a few commits since I found this one.

I was indeed using that command.

**Comment by hamishmb** [comment]
My laptop's CPU fan has just died, so it'll be a few days before I can fire up the VM again to check, ah well.

**Comment by hamishmb** [comment]
Okay, fixed it, updated and rebuilt Darling with the latest changes as instructed in the build instructions.

I still have the crash. I did a debug build though, so when I have some more time I'll report back with some debugging results.

**Comment by facekapow** [comment]
Are you still able to reproduce this with the latest version of Darling?

**Comment by hamishmb** [comment]
Yep, still having this issue with Debian 12 and Python 3.10.11, xtrace log incoming.

**Comment by facekapow** [comment]
Thanks; it seems that what's crashing isn't actually the `installer` command but rather a postinstall script:

```
[87] execve("./postinstall", {"./postinstall", "Downloads/python-3.10.11-macos11.pkg", "/Library/Frameworks/Python.framework", "/"}, {"XDG_VTNR=7", "MANPATH=/usr/local/share/man:", "SSH_AGENT_PID=1313", "XDG_SESSION_ID=3", "XDG_DATA_HOME=/home/hamish/.local/share", "GPG_AGENT_INFO=/run/user/1000/gnupg/S.gpg-agent:0:1", "SHELL=/bin/bash", "XDG_MENU_PREFIX=lxqt-", "TERM=xterm-256 -> 0
[85] close(4) -> 0
[85] wait4(87, 0x7FFFFFDFF30C, 0, NULL)
```

When a parent process calls `wait4(2)` on a child and that crashed, the signal gets sent to the parent as well.

**Comment by facekapow** [comment]
I unfortunately still can't reproduce this on my end. Could you perhaps try this again in a new prefix? You can either [set `DPREFIX`](https://docs.darlinghq.org/darling-prefix.html#darling-prefix) before starting Darling or delete your current prefix. This time, before starting Darling, run the following:

```bash
darling shutdown # just in case
# if you're using the default prefix, replace `${DPREFIX}` here with `${HOME}/.darling`
rm -f ${DPREFIX}/private/var/log/dserver.log
export DSERVER_LOG_LEVEL=info
export XTRACE_NO_COLOR=1
export XTRACE_KPRINTF=1
```

Then just run the command with `xtrace` (as you did), but you should get the normal command output. Once it crashes and you exit Darling, please post the contents of `${DPREFIX}/private/var/log/dserver.log`.

**Comment by hamishmb** [comment]
Okay so now I'm getting different behaviour. It either just seems to freeze up, or I get lots of:

`pthread_cond_signal: Resource temporarily unavailable`

and:

`dyld: warning: could not load inserted library '/usr/lib/darling/libxtrace.dylib,' iinto hardened process because image not found`

Messages. If I wait for long enough, it does seem like it's making progress though. Eventually, it crashed with Abort trap: 6 (core dumped). I'll have to attach the log later because I don't have a way of getting it off of my VM's HDD at the moment.

**Comment by hamishmb** [comment]
Apologies for the delay, here's the log file. It's huge uncompressed, but hopefully what's in there will be helpful.
[dserver.log.zip](https://github.com/darlinghq/darling/files/13416063/dserver.log.zip)

**Comment by hamishmb** [comment]
Just a note to say I'm going to try to reproduce this again, probably some time this week, just to see if I'm still getting the same error.

**Comment by hamishmb** [comment]
Okay, closing as I can no longer reproduce this issue with either Python 3.10.5 or the latest version at the time of writing.
